strapi/strapi
# chore: bump glob from 7.2.0 to 10.4.2 in core/strapi

### What does it do?

This pull request updates the `glob` dependency in Strapi from version 7.2.0 to 10.4.2. The reason for this change is that versions of `glob` prior to 9 are deprecated and no longer supported (https://www.npmjs.com/package/glob/v/7.2.0?activeTab=code). Additionally, the older version includes the `inflight` package, which has a known vulnerability (https://github.com/isaacs/inflight-DEPRECATED-DO-NOT-USE/issues/18, https://security.snyk.io/package/npm/inflight/1.0.6).

### Why is it needed?

Updating `glob` to version 10.4.2 resolves the deprecated status and eliminates the vulnerability associated with the `inflight` package, thereby enhancing the security and maintainability of the Strapi package.

### How to test it?

To verify the behavior:
- Ensure you have set up a testing environment similar to the one specified in Strapi's documentation.
- Execute existing tests and ensure they pass without errors related to the `glob` functionality.
- Verify that the `glob` package functions correctly in scenarios where it is utilized within Strapi.

### Related issue(s)/PR(s)

This pull request is related to fixing the issues with deprecated `glob` versions and the `inflight` vulnerability.

(https://github.com/isaacs/inflight-DEPRECATED-DO-NOT-USE/issues/18, https://security.snyk.io/package/npm/inflight/1.0.6)
